cmds/core/md5sum/md5sum.go

```diff
@@ -7,84 +7,82 @@ package main
 
 import (
 	"crypto/md5"
+	"flag"
 	"fmt"
 	"io"
 	"log"
 	"os"
 
-	"github.com/spf13/pflag"
+	"github.com/u-root/u-root/pkg/uroot/util"
 )
 
-func getInput() (input []byte, err error) {
-	return io.ReadAll(os.Stdin)
-}
-
-func helpPrinter() {
-	fmt.Printf("Usage:\nmd5sum <File Name>\n")
-	pflag.PrintDefaults()
-	os.Exit(0)
-}
+var (
+	help = flag.Bool("h", false, "Show this help and exit")
+)
 
-func versionPrinter() {
-	fmt.Println("md5sum utility, URoot Version.")
-	os.Exit(0)
-}
+var usage = "md5sum: md5sum <File Name>"
 
-func calculateMd5Sum(fileName string, data []byte) string {
+func calculateMd5Sum(w io.Writer, fileName string, data []byte) (string, error) {
```

Comment:
I don't get this function at all. It always returns an empty string, it takes file names and data, and it prints to an io.Writer? Could we, perhaps, simplify this? Also, what happens on a 512 Mib System with a 2 GiB file? The package is not being used correctly.

calculateMD5Sum should take an io.Reader, and return a ([]byte, error). You can then do this:

func calculateMd5Sum(r io.Reader) ([]byte, error) {
	h := md5.New()
	if _, err := io.Copy(h, r); err != nil {
		log.Fatal(err)
	}
return h.Sum(nil)
}

we don't need chaining, but chaining works too: https://go.dev/play/p/RCVn197QXX4

From here, the rest of the program can be changed. The basic structure of this program can be very similar to cat.

Then we can rewrite it all with generics start to come out :-)

Reply:
A proposal: I removed the function and the function did not grow noteworthy. I think the code is now more simple. What do your think?